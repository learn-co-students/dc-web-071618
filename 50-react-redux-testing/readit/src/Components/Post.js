import React from "react";

const Post = ({ post }) => {
  return (
    <div className="postComponent">
      <a href={post.url}>
        <h3>{post.title}</h3>
      </a>
      <p> posted_by: {post.author}</p>
    </div>
  );
};

export default Post;
