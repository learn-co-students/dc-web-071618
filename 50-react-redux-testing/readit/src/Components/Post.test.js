import React from "react";
import { shallow } from "enzyme";

import Post from "./Post";

let dummyPost = {
  title: "Im a dummy",
  author: "A real dummy",
  url: "www.google.com"
};

describe("Post", () => {
  it("matches its snapshot", () => {
    let wrapper = shallow(<Post post={dummyPost} />);
    expect(wrapper).toMatchSnapshot();
  });
});
