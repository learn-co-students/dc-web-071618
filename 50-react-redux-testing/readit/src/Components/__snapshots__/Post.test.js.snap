// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches its snapshot 1`] = `
<div
  className="postComponent"
>
  <a
    href="www.google.com"
  >
    <h3>
      Im a dummy
    </h3>
  </a>
  <p>
     posted_by: 
    A real dummy
  </p>
</div>
`;
